File: algo/reverse-bits/reverseBits.py
class Solution(object):
    def reverseBits(self, n):
        """
        :type n: int
        :rtype: int
        """
        rvs, cnt = 0, 32
        while cnt > 0:
            cnt -= 1
            rvs <<= 1
            rvs |= (0x1 & n)
            n >>= 1
        return rvs

File: algo/reverse-bits/reverseBits.c
uint32_t reverseBits(uint32_t n) {
    uint32_t rvs;
    int cnt;
    
    cnt = 32;
    rvs = 0;
    while (cnt-- > 0) {
        rvs <<= 1;
        rvs |= (n & 0x1);
        n >>= 1;
    }
    return rvs;
}